plugins/woocommerce/includes/class-wc-order.php

```diff
@@ -1074,6 +1074,56 @@ protected function set_address_prop( $prop, $address, $value ) {
 		}
 	}
 
+	/**
+	 * Setter for billing address, expects the $address parameter to be key value pairs for individual address props.
+	 *
+	 * @param array $address Address to set.
+	 *
+	 * @return void
+	 */
+	public function set_billing_address( array $address ) {
+		foreach ( $address as $key => $value ) {
+			$this->set_address_prop( $key, 'billing', $value );
+		}
+	}
+
+	/**
+	 * Shortcut for calling set_billing_address.
+	 *
+	 * This is useful in scenarios where set_$prop_name is invoked, and since we store the billing address as 'billing' prop in data, it can be called directly.
+	 *
+	 * @param array $address Address to set.
+	 *
+	 * @return void
+	 */
+	public function set_billing( array $address ) {
```

Comment:
I don't think`__set` will support the `set_$prop_name` scenarios, seems like it's for handling directly setting class props. 

I know it's a bit confusing, but we do have a prop called `billing` which is stored directly as part of the order object. In that context, `set_billing` does make sense since its setting that specific prop. wdyt?

Reply:
Ok, still not a big fan, but as long as we document it properly and it's consistent with our approach for other property setting methods, it's fine.